src/ray/object_manager/object_manager.cc

```diff
@@ -332,57 +332,60 @@ void ObjectManager::HandleReceiveFinished(const ObjectID &object_id,
 void ObjectManager::Push(const ObjectID &object_id, const NodeID &node_id) {
   RAY_LOG(DEBUG) << "Push on " << self_node_id_ << " to " << node_id << " of object "
                  << object_id;
-  if (local_objects_.count(object_id) == 0) {
-    // Avoid setting duplicated timer for the same object and node pair.
-    auto &nodes = unfulfilled_push_requests_[object_id];
-    // Issue a restore request if the object is on local disk. This is only relevant
-    // if the local filesystem storage type is being used.
-    auto object_url = get_spilled_object_url_(object_id);
-    if (!object_url.empty()) {
-      restore_spilled_object_(object_id, object_url, nullptr);
-    }
-    if (nodes.count(node_id) == 0) {
-      // If config_.push_timeout_ms < 0, we give an empty timer
-      // and the task will be kept infinitely.
-      std::unique_ptr<boost::asio::deadline_timer> timer;
-      if (config_.push_timeout_ms == 0) {
```

Comment:
Some of this logic still is required for timeout handling? Any reason why we deleted all?

Reply:
oh you are right... actually i was confused by what this tries to achieve...

Reply:
that's actually a good point! I thought it only deals with spilled objects.

Reply:
huh, read the logic again. After this PR we shouldn't really differentiate the difference between plasma/local spilled object? If this is the case, removing this logic makes sense?

Reply:
Isn't it rather supposed to be that both are needed to be supported? I don't understand if other changes could replace the logic this code block uses to have. @stephanie-wang Is this block okay to remove?

Reply:
It seems there are three cases we need to handle:
- Object is in-memory locally.
- Object is spilled locally.
- Object is not yet created (pending creation).

I think this PR is breaking the last case. This caller will eventually retry the push request, so this might not manifest itself as a correctness issue, but will definitely impact performance. Ray issues push requests pre-emptively before objects have been created in certain cases I believe, for example for actor tasks (cc @stephanie-wang ).